src/SimpleAmqpClient/Channel.h

```diff
@@ -191,6 +191,12 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable {
  public:
   virtual ~Channel();
 
+  /**
+    * Exposes the underlying socket handle
+    * @returns file descriptor number associated with the connection socket
```

Comment:
Please add a warning to the doc string:

```
WARNING: this function exposes an internal implementation detail SimpleAmqpClient.
Manipulating the socket descriptor will result in undefined behavior of the library.
Additionally SimpleAmqpClient's use of the socket will change depending on what
version of rabbitmq-c and SimpleAmqpClient are used. Test carefully before depending
on any specific behavior.
```

(please reflow the above to 80 character columns within the comment).

Reply:
I could argue whether socket ownership is an implementation detail at all: the API specifies (in prose, informally) that it *makes a connection* to the AMQP broker, among other things. Could any library make a TCP connection without a socket and an FD?..

I take your point nonetheless; the disclaimer added as [Doxygen \warning block][•].

[•]: https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdwarning

Reply:
Implementation detail may be too strong of a statement.  The point I'm trying to get across is two-fold:

1.  Users of the API should should not read or write (or otherwise change the state) of the socket returned from this function.  Doing so will result in undefined behavior of the library.

2. User should not rely on a set behavior of the socket being returned.  The library may change how it uses the socket from release to release.  E.g., a future release of SimpleAmqpClient may do automatic re-connection, so the socket may become invalid between calls, or the library may choose to open multiple TCP connections to the broker to improve throughput.

I think what you've got captures that sufficiently.